src/compiler/checker.ts

```diff
@@ -11507,7 +11507,7 @@ namespace ts {
                 if (symbol === globalThisSymbol) {
                     const varsOnly = new Map<string, Symbol>() as SymbolTable;
                     members.forEach(p => {
-                        if (!(p.flags & SymbolFlags.BlockScoped)) {
+                        if (!(p.flags & SymbolFlags.BlockScoped) && !(p.flags & SymbolFlags.ValueModule && some(p.declarations, isAmbientModule))) {

```

Comment:
some questions: 
- what's this `ValueModule` symbol flag/what is a value module?
- is it precise enough to check if some declaration is an ambient module? what if some of the other declarations are not ambient modules? (that question may be answered by the one above)

Reply:
* A value module is a namespace OR `declare module "foo"` with values inside it. A namespace module is a namespace that has only types inside it. (Both kinds of module can contain other namespaces; any nested value turns the parent namespaces into value modules.)
* I think the binder only allows ambient modules to merge with ambient modules? I'll have to check.

Reply:
Hm, the excludes are the same as other value modules: `Value & ~(Function | Class  | RegularEnum | ValueModule)`

I'll see if I can name a function, class or enum `"foo"`, but I don't think I can.

Reply:
So, effectively only other ambient modules can merge with ambient modules.

Reply:
> I'll see if I can name a function, class or enum "foo", but I don't think I can.

You can if you use a js merge, eg,

```ts
globalThis['"module"'] = function () {}
```

or similar.